src/payment/cybersource/data/sagas.js

```diff
@@ -0,0 +1,60 @@
+import { call, put, select, takeEvery } from 'redux-saga/effects';
+import {
+  SUBMIT_PAYMENT_CYBERSOURCE,
+  submitPaymentCybersourceBegin,
+  submitPaymentCybersourceFailure,
+} from './actions';
+
+import * as CybersourceApiService from './service';
+
+import { generateAndSubmitForm } from '../../../common/utils';
+import { handleErrors } from '../../../feedback';
+import { configuration } from '../../../environment';
+
+export function* handleSubmitPaymentCybersource(action) {
+  const { cardHolderInfo, cardDetails } = action.payload;
+  yield put(submitPaymentCybersourceBegin());
+  try {
+    const sdnCheck = yield call(
+      CybersourceApiService.sdnCheck,
+      cardHolderInfo.firstName,
+      cardHolderInfo.lastName,
+      cardHolderInfo.city,
+      cardHolderInfo.country,
+    );
+
+    if (sdnCheck.hits > 0) {
+      window.location.href = `${configuration.ECOMMERCE_BASE_URL}/payment/sdn/failure/`;
+    }
+
+    const basket = yield select(state => ({ ...state.payment.basket }));
+    const checkout = yield call(
```

Comment:
May be nice in the future to add comments here that describe checkout vs. generateAndSubmitForm below